tree/30_generic_methods/file_edit_line_key_value_ensure_parameter_in_list.cf

```diff
@@ -0,0 +1,113 @@
+#####################################################################################
+# Copyright 2017 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+# @name File edit line key value ensure parameter in list
+# @description Ensure that one parameter exists in a list of parameters, on one single line, in the right hand side of a key->values line
+# @documentation Edit the file, and ensure it contains the defined parameter in the list of values on the right hand side of a key->values line.
+# If the parameter is not there, it will be added at the end, separated by parameter_separator.
+# Optionnaly, you can define leading and closing character to enclose the parameters
+# If the lhs does not exist in the file, it will report an error
```

Comment:
This seems surprising to me, I would expect this method to add the line, because we have everything we need to do it.